File: src/python/fods/spec/table/covered_table_cell.py
"""
ODF spec element: table:covered-table-cell

Spec ref: ODF 1.3 §9.5.2 — Covered Table Cell
Fact ref: SAL-FODS-00023
QName: table:covered-table-cell
Namespace: urn:oasis:names:tc:opendocument:xmlns:table:1.0
Canonical class: Table.CoveredTableCell
"""

from typing import ClassVar


class CoveredTableCell:
    """Canonical spec-shaped class for table:covered-table-cell in FODS context.

    A marker element for cells that are covered by a spanning table cell
    (table:table-cell with table:number-columns-spanned or table:number-rows-spanned).
    Has the same attributes as table:table-cell but no content is rendered.

    This is NOT the production model. Facade: use Compat/ layer for production access.
    """

    spec_qname: ClassVar[str] = "table:covered-table-cell"
    spec_fact_ref: ClassVar[str] = "SAL-FODS-00023"
    namespace_uri: ClassVar[str] = "urn:oasis:names:tc:opendocument:xmlns:table:1.0"
    local_name: ClassVar[str] = "covered-table-cell"
    spec_source: ClassVar[str] = "ODF 1.3 Part 3"
    facade_names: ClassVar[list] = []

File: src/net/fods/Model/Table/CoveredTableCell.cs
// FormatFactory.Fods — Model.Table.CoveredTableCell
// spec_qname: table:covered-table-cell
// spec_fact_ref: FACT-FODS-007
// Authority: plans/.claude/imperative-drifting-conway.md §2
// TC-W1-FODS-NET-001

namespace FormatFactory.Fods.Model.Table;

/// <summary>
/// Canonical runtime model for the ODF table:covered-table-cell element.
///
/// ODF 1.3 §9.4.6 — table:covered-table-cell represents a cell that is covered
/// by a spanning cell in a merged range.
/// spec_qname: table:covered-table-cell
/// spec_fact_ref: FACT-FODS-007
/// </summary>
public sealed class CoveredTableCell
{
    /// <summary>The ODF QName for this element.</summary>
    public const string SpecQName = "table:covered-table-cell";

    /// <summary>The SAL fact reference (sal-facts-latest.json).</summary>
    public const string SpecFactRef = "FACT-FODS-007";

    /// <summary>table:style-name attribute — the covered cell's style.</summary>
    public string? StyleName { get; set; }

    /// <summary>
    /// table:number-columns-repeated — repeat count for empty covered cells.
    /// Default: 1.
    /// </summary>
    public int RepeatCount { get; set; } = 1;
}
